store/store.go

```diff
@@ -0,0 +1,180 @@
+package store
+
+import (
+	"time"
+
+	"github.com/tidepool-org/platform/Godeps/_workspace/src/labix.org/v2/mgo"
+	"github.com/tidepool-org/platform/Godeps/_workspace/src/labix.org/v2/mgo/bson"
+
+	"github.com/tidepool-org/platform/config"
+	"github.com/tidepool-org/platform/logger"
+)
+
+var (
+	log = logger.Log.GetNamed("store")
+)
+
+//Store is a generic store interface for all operations that we need
+// Note: there will be a specific implementation for the store
+// type we are using, which could also be a mock
+type Store interface {
+	Save(d interface{}) error
+	Update(selector interface{}, d interface{}) error
+	Delete(find Fields) error
+	Read(find Fields, remove Fields, result interface{}) error
+	ReadAll(find Fields, remove Fields) Iterator
+}
+
+//Iterator for the query iterator
+type Iterator interface {
+	Next(result interface{}) bool
+	Close() error
+}
+
+//ClosingSessionIterator so we can manage the session for our iterator
+type ClosingSessionIterator struct {
+	*mgo.Session
+	*mgo.Iter
+}
+
+//Fields for finding or updating an item
+type Fields map[string]interface{}
+
+//MongoStore is the mongo implementation of Store
+type MongoStore struct {
+	Session        *mgo.Session
+	CollectionName string
+	Config         MongoConfig
+}
+
+//MongoConfig is the required config for the MongoStore
+type MongoConfig struct {
+	URL     string `json:"connectionUrl"`
+	DbName  string `json:"databaseName"`
+	Timeout int    `json:"timeout"`
+}
+
+//NewMongoStore returns an initailised instance of MongoStore
+func NewMongoStore(name string) *MongoStore {
+
+	store := &MongoStore{CollectionName: name}
+	config.FromJSON(&store.Config, "mongo.json")
```

Comment:
Inject config in here rather than reading it here would make for easier testing?